File: android/app/src/main/java/com/adqmobile/socialnetwork/domain/post/PostRepository.kt
package com.adqmobile.socialnetwork.domain.post

interface PostRepository {
    suspend fun createPost(post: Post)
    suspend fun reactToPost(reaction: Reaction)
    suspend fun retrieveAllPosts(): Array<Post>
    suspend fun retrieveReactionsForPost(post: Post): Array<Reaction>
}

File: ios/socialnetwork/domain/post/PostRepository.swift
//
//  File.swift
//  socialnetwork
//
//  Created by Alan Donizete Quintiliano on 13/12/19.
//  Copyright © 2019 Alan Donizete Quintiliano. All rights reserved.
//

import Combine

protocol PostRepository {
    func storePost(post: Post) -> AnyPublisher<Post, Error>

    func createPost(post: Post) -> AnyPublisher<Post, Error>

    func getPosts() -> AnyPublisher<[Post], Error>

    func readPosts() -> AnyPublisher<[Post], Error>

    func storeReaction(reaction: Reaction) -> AnyPublisher<Reaction, Error>

    func reactToPost(reaction: Reaction) -> AnyPublisher<Post, Error>
}
